x-pack/plugins/aiops/public/components/spike_analysis_table/links_menu.tsx

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { escapeKuery } from '@kbn/es-query';
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import type { ChangePoint } from '@kbn/ml-agg-utils';
+import {
+  EuiButtonIcon,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiPopover,
+  EuiProgress,
+  EuiToolTip,
+} from '@elastic/eui';
+import { useAiOpsKibana } from '../../kibana_context';
+import { SEARCH_QUERY_LANGUAGE } from '../../application/utils/search_utils';
+
+interface LinksMenuProps {
+  changePoint: ChangePoint;
+  dataViewId?: string;
+}
+
+export const LinksMenu: FC<LinksMenuProps> = ({ changePoint, dataViewId }) => {
+  const [isPopoverOpen, setPopoverOpen] = useState(false);
+  const onButtonClick = setPopoverOpen.bind(null, !isPopoverOpen);
+  const closePopover = setPopoverOpen.bind(null, false);
+  const [openInDiscoverUrl, setOpenInDiscoverUrl] = useState<string | undefined>();
+  const [discoverUrlError, setDiscoverUrlError] = useState<string | undefined>();
+
+  const aiOpsKibana = useAiOpsKibana();
+  const {
+    services: { application, share, data },
+  } = aiOpsKibana;
+
+  useEffect(() => {
+    let unmounted = false;
+
+    if (!application.capabilities.discover?.show) {
+      const discoverNotEnabled = i18n.translate(
+        'xpack.ml.anomaliesTable.linksMenu.discoverNotEnabledErrorMessage',
+        {
+          defaultMessage: 'Discover is not enabled',
+        }
+      );
+
+      if (!unmounted) {
+        setDiscoverUrlError(discoverNotEnabled);
+      }
+      return;
+    }
+
+    const discoverLocator = share.url.locators.get('DISCOVER_APP_LOCATOR');
+    if (!discoverLocator) {
+      const discoverLocatorMissing = i18n.translate(
+        'xpack.ml.anomaliesTable.linksMenu.discoverLocatorMissingErrorMessage',
+        {
+          defaultMessage: 'No locator for Discover detected',
+        }
+      );
+
+      if (!unmounted) {
+        setDiscoverUrlError(discoverLocatorMissing);
+      }
+      return;
+    }
+
+    if (!dataViewId && !unmounted) {
```

Comment:
Will this ever be hit, as you can only come into the view after picking a data view?

Reply:
Good call - removed in 0c14669335f451c5a2623f91b9b65158005da401